src/store/ui/reducer.tsx

```diff
@@ -1,18 +1,45 @@
 import { createReducer } from '@reduxjs/toolkit';
-import { doAddToast, doRemoveToast, appTime } from '@store/ui/actions';
+import { doAddToast, doRemoveToast, appTime, setNetworks, selectNetwork } from '@store/ui/actions';
 import { ToastType } from '@blockstack/ui';
 
-interface UiState {
+export interface Network {
+  MOCKNET?: string;
+  TESTNET?: string;
+  LOCAL?: 'http://localhost:3999';
+}
+export type NetworkOptions = 'MOCKNET' | 'TESTNET' | 'LOCAL';
+export interface UiState {
   toasts: ToastType[];
   appTime: number;
+  config: {
+    network: Network;
+    selectedNetwork: NetworkOptions;
+  };
 }
 
 const initialState: UiState = {
   toasts: [],
   appTime: Math.round(new Date().getTime() / 1000),
+  config: {
+    network: {
+      MOCKNET: undefined,
+      TESTNET: undefined,
+      LOCAL: 'http://localhost:3999',
+    },
+    selectedNetwork: 'MOCKNET',
+  },
 };
 
 export const uiReducer = createReducer(initialState, builder => {
+  builder.addCase(setNetworks, (state, action) => {
+    state.config.network = {
+      ...state.config.network,
+      ...action.payload,
+    };
+  });
```

Comment:
I see this is part of redux-toolkit (checked docs) I find it super counterintuitive to set the state object like this rather than returning an object.

A wider point here: IMO it's a super poor API decision on their part to facilitate this. A reader of this code has to understand a pretty specific implementation of redux-toolkit to know that this isn't mutating state (even if they're familiar with the redux pattern). And what's the uptake? A few cases where some lines of code are saved? I know using too many spred operators gets verbose, but that verbosity is worth not breaking a pattern as widely understood as redux.

There are perfectly good alternatives too, such as the [lens pattern](https://functional.christmas/2019/6), or using `Immer` explicitly (and not behind redux-toolkits implementation). 

Thoughts? Would really love it if we could avoid mutating state like this.

Reply:
I think this is pretty subjective and open to opinion. It's more of a code style thing (one that I am very much in favor of, and so are 16k other folks using immer). 

It's not a true mutation, but I understand the concern. I think we could in a different time talk about the pros and cons of using this toolset in the way we are, but it's very much out of scope for this PR.